internal/models/reservation_model.go

```diff
@@ -80,6 +80,41 @@ type AWSReservation struct {
 	Detail *AWSDetail `db:"detail" json:"detail"`
 }
 
+type GCPDetail struct {
+	Zone string `json:"zone"`
+
+	// Optional instance name
+	Name *string `json:"name"`
+
+	// GCP Machine type.
+	MachineType string `json:"machine_type"`
+
+	// Amount of instances to provision of type: Instance type.
+	Amount int64 `json:"amount"`
+
+	// Immediately power off the system after initialization
+	PowerOff bool `json:"poweroff"`
+}
```

Comment:
Important observation: You are missing Region, but you have a Zone. Zones live within a Region, I think we should support both from the very beginning.

Reply:
it does not seem like there is a region specification when creating an instance https://github.com/googleapis/go-genproto/blob/18f8339a66a5/googleapis/cloud/compute/v1/compute.pb.go#L27032